### PR TITLE
Rc2021 anv

### PR DESCRIPTION
**Pull request by tonnov**
Mini-Challenge 4: State Management

### CODE REVIEW

File: src/providers/Global/Global.provider.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        return {
+          ...state,
+          darkTheme: !state.darkTheme
+        }
+    default:
+      return state;
+  }
+}
+
+export const GlobalProvider = (props) => {
```
Comment (andresSaldanaAguilar) [inline src/providers/Global/Global.provider.jsx:43]: nice reducer and provider Antonio 👍

File: src/providers/Global/Global.provider.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    });
+
+
+    it('should toggle the darkTheme state', () => {
+        const { result } = renderHook(() => useGlobal(), { wrapper } );
+
+        const { dispatch } = result.current;
+
+        act(() => {
+            dispatch({ type: 'toggle_theme', payload: true });
```
Comment (andresSaldanaAguilar) [inline src/providers/Global/Global.provider.spec.jsx:35]: en este caso necesitamos payload?

File: src/providers/Global/Global.provider.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        })
+
```
Comment (andresSaldanaAguilar) [inline src/providers/Global/Global.provider.spec.jsx:37]: quiza para probar el funcionamiento podrias checar que el estado inicial de darktheme se niegue

**Comment by andresSaldanaAguilar** [comment]
Good job Antonio! nice use of use reducer and context, just left some comments, let me know if I can help you with reducer tests, I saw your test for it correct
**for the following PRs, create a branch from the work done and not from the start of the project, because it is more difficult to figure out what was added on this specific PR**

## Acceptance Criteria
- [x] The search term is stored and retrieved from the Global Context correctly.
- [x] The appearance theme is stored on the Global Context and applied correctly to the App UI.
- [x] `useReducer` hook is implemented correctly to manage the Global State.

## Bonus Points
- [ ] Testing coverage is above 70%. Coverage not provided.

File: src/providers/Global/Global.provider.spec.jsx (hunk 2 above)
Comment (tonnov) [reply src/providers/Global/Global.provider.spec.jsx:35]: Not really, it just toggle the boolean 😬

**Pull request merged**